lib/spack/spack/paths.py

```diff
@@ -86,7 +86,11 @@
 # setting `SPACK_USER_CACHE_PATH`. Otherwise it defaults to ~/.spack.
 #
 def _get_user_cache_path():
-    return os.path.expanduser(os.getenv("SPACK_USER_CACHE_PATH") or "~%s.spack" % os.sep)
+    return (
+        var_path
+        if "SPACK_DISABLE_LOCAL_CONFIG" in os.environ
+        else os.path.expanduser(os.getenv("SPACK_USER_CACHE_PATH") or "~%s.spack" % os.sep)
```

Comment:
This logic changes behavior when both SPACK_USER_CACHE_PATH and SPACK_DISABLE_LOCAL_CONFIG are set. Previously it would use SPACK_USER_CACHE_PATH, now it will use var_path.

Reply:
You are right that the behaviour changed if user intentionally set both. Will fix.


I would still argue that the env name`SPACK_USER_CACHE_PATH` is confusing, for the reason it has the word `USER` is suggestive it has nothing to do at site level.

We may consider change this when fixing the todo few lines above the current change.
e.g. separate another env for site like `SPACK_SITE_CACHE_PATH`